clients/git/e2e_test.go

```diff
@@ -15,41 +15,61 @@
 package git
 
 import (
+	"strings"
+
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 
 	"github.com/ossf/scorecard/v4/clients"
+	"github.com/ossf/scorecard/v4/clients/githubrepo"
+	"github.com/ossf/scorecard/v4/clients/gitlabrepo"
+	"github.com/ossf/scorecard/v4/clients/localdir"
 )
 
 var _ = DescribeTable("Test ListCommits commit-depth for HEAD",
 	func(uri string) {
 		const commitSHA = clients.HeadSHA
 		const commitDepth = 1
 		client := &Client{}
-		Expect(client.InitRepo(uri, commitSHA, commitDepth)).To(BeNil())
+		GinkgoT().Logf("URI: %s\n", uri)
+		repo, err := getRepoClient(uri)
+		Expect(err).To(BeNil())
+		Expect(client.InitRepo(repo, commitSHA, commitDepth)).To(BeNil())
 		commits, err := client.ListCommits()
 		Expect(err).To(BeNil())
 		Expect(len(commits)).Should(BeEquivalentTo(commitDepth))
 		Expect(client.Close()).To(BeNil())
 	},
+	Entry("Local", "../../"),
 	Entry("GitHub", "https://github.com/ossf/scorecard"),
-	Entry("Local", "file://../../"),
 	Entry("GitLab", "https://gitlab.haskell.org/haskell/filepath"),
 )
 
+func getRepoClient(uri string) (clients.Repo, error) {
+	if strings.Contains(uri, "github.com") {
+		return githubrepo.MakeGithubRepo(uri)
+	} else if strings.Contains(uri, "gitlab") {
+		return gitlabrepo.MakeGitlabRepo(uri)
+	}
+	return localdir.MakeLocalDirRepo(uri)
+}
+
 var _ = DescribeTable("Test ListCommits commit-depth and latest commit at [0]",
 	func(uri, commitSHA string) {
 		const commitDepth = 10
 		client := &Client{}
-		Expect(client.InitRepo(uri, commitSHA, commitDepth)).To(BeNil())
+		repo, err := getRepoClient(uri)
+		GinkgoT().Logf("URI: %s\n", uri)
+		Expect(err).To(BeNil())
+		Expect(client.InitRepo(repo, commitSHA, commitDepth)).To(BeNil())
 		commits, err := client.ListCommits()
 		Expect(err).To(BeNil())
 		Expect(len(commits)).Should(BeEquivalentTo(commitDepth))
 		Expect(commits[0].SHA).Should(BeEquivalentTo(commitSHA))
 		Expect(client.Close()).To(BeNil())
 	},
-	Entry("GitHub", "https://github.com/ossf/scorecard", "c06ac740cc49fea404c54c036000731d5ea6ebe3"),
-	Entry("Local", "file://../../", "c06ac740cc49fea404c54c036000731d5ea6ebe3"),
+	Entry("https://github.com/ossf/scorecard", "c06ac740cc49fea404c54c036000731d5ea6ebe3"),
```

Comment:
Should this entry have "GitHub" out front?

Reply:
Took care of it. Thanks